javascripts/content_scripts/privly.js

```diff
@@ -564,16 +564,21 @@ var privly = {
 
     "use strict";
 
+            // typeof(message.data.frame_id) != "string"
+
     //check the format of the message
-    if (typeof(message.origin) !== "string" ||
-        typeof(message.data) !== "string" ||
-        message.data.indexOf(',') < 1) {
-      return;
+     if (typeof(message.origin) !== "string" ){
+        return;
     }
 
+    var msg = JSON.parse(message.data);
+
     //Get the element by name.
-    var data = message.data.split(",");
-    var iframe = document.getElementsByName(data[0])[0];
+    var frameName = msg.frame_id;
```

Comment:
Many messages are sent between many different contexts. Here you should also check that the message you are expecting was sent by looking at the keys in the message. You should also handle improperly formatted JSON.